### DISCUSSION TITLE
How do i target specific module

### DISCUSSION THREAD

**Discussion by harrypotter033** [aztfmod/terraform-azurerm-caf#1326] [Q&A]
`compute = {
    virtual_machine_scale_sets = {
      api-vmss = {
        resource_group_key = module.variables.rg["blue"]
        os_type            = "linux"
        # keyvault_key                         = "vault_uri"
        public_key_pem_file = "sshdev.pub"

        vmss_settings = {
          linux = module.vmss_api.vmss_settings_linux

        }

        network_interfaces = {
          nic0 = module.vmss_api.network_interfaces
        }

        data_disks = {
          data1 = module.vmss_api.data_disks
        }

      }

      web-vmss = {
        resource_group_key = module.variables.rg["blue"]
        os_type            = "linux"
        # keyvault_key                         = "vault_uri"
        public_key_pem_file = "ssh-dev.pub"

        vmss_settings = {
          linux = module.vmss_web.vmss_settings_linux

        }

        network_interfaces = {
          nic0 = module.vmss_web.network_interfaces
        }

        data_disks = {
          data1 = module.vmss_web.data_disks
        }

      }
    }
`
I tried to to build only specif module 
![image](https://user-images.githubusercontent.com/4506144/188037136-ec97dd60-c4c9-4863-a4fa-4564a2c56b3c.png)
![image](https://user-images.githubusercontent.com/4506144/188037183-f2c71b5a-2ebb-4290-bbe1-9dc13772ca68.png)
In terraform apply command but its not working.

**Comment by brk3** [comment]
Here's an example that creates a resource group, and a vnet:

```
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
    }
  }
}

provider "azurerm" {
  features {}
}

provider "azurerm" {
  features {}
  alias = "vhub"
}

module "caf" {
  source = "aztfmod/caf/azurerm"

  providers = {
    azurerm.vhub = azurerm.vhub
  }

  global_settings = {
    default_region = "region1"
    regions = {
      region1 = "northeurope"
    }
  }

  resource_groups = {
    test = {
      name = "test"
    }
  }

  # az ad signed-in-user show --query id -o tsv
  logged_user_objectId = "..."

  networking = {
    vnets = {
      vnet1 = {
        vnet = {
          name          = "test-vn"
          address_space = ["10.2.0.0/16"]
        }
        subnets = {}
        resource_group_key = "test"
      }
    }
  }
```

If I wanted to just create the resource group and not the vnet, I can target it like so:

```
terraform plan -target module.caf.module.resource_groups
```
